cmd/logs.go

```diff
@@ -26,106 +28,149 @@ import (
 	"github.com/spf13/cobra"
 )
 
+const (
+	maxEsLogsPerQuery     = 10000
+	fourteenDaysInSeconds = 60 * 60 * 24 * 14
+)
+
 // numberOfLines of logfile to return
 var numberOfLines = "400"
 
-// logsCmd represents the logs command
-var logsCmd = &cobra.Command{
-	Use:   "logs (gardener-apiserver|gardener-controller-manager|ui|api|scheduler|controller-manager|etcd-operator|etcd-main[etcd backup-restore]|etcd-main-backup|etcd-events[etcd backup-restore]|addon-manager|vpn-seed|vpn-shoot|machine-controller-manager|dashboard|prometheus|grafana|alertmanager|tf (infra|dns|ingress)",
-	Short: "Show and optionally follow logs of given component\n",
-	Long:  `s`,
-	Run: func(cmd *cobra.Command, args []string) {
-		if len(args) < 1 || len(args) > 2 {
-			fmt.Println("Command must be in the format: logs (gardener-apiserver|gardener-controller-manager|ui|api|scheduler|controller-manager|etcd-operator|etcd-main[etcd backup-restore]|etcd-events[etcd backup-restore]|addon-manager|vpn-seed|vpn-shoot|machine-controller-manager|dashboard|prometheus|grafana|alertmanager|tf (infra|dns|ingress)")
+//flags passed to the command
+var flags *logFlags
+
+func newCmdLogs() *cobra.Command {
+	flags = newLogsFlags()
+	cmd := &cobra.Command{
+		Use:   "logs (gardener-apiserver|gardener-controller-manager|ui|api|scheduler|controller-manager|etcd-operator|etcd-main[etcd backup-restore]|etcd-main-backup|etcd-events[etcd backup-restore]|addon-manager|vpn-seed|vpn-shoot|machine-controller-manager|dashboard|prometheus|grafana|alertmanager|tf (infra|dns|ingress)",
+		Short: "Show and optionally follow logs of given component\n",
+		Long:  `s`,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			validateArgs(args)
+			validateFlags(flags)
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			runCommand(args)
+		},
+		ValidArgs: []string{"gardener-apiserver", "gardener-controller-manager", "ui", "api", "scheduler", "controller-manager", "etcd-operator", "etcd-main", "etcd-events", "addon-manager", "vpn-seed", "vpn-shoot", "auto-node-repair", "dashboard", "prometheus", "grafana", "alertmanager", "tf"},
+		Aliases:   []string{"log"},
+	}
+	cmd.Flags().Int64Var(&flags.tail, "tail", flags.tail, "Lines of recent log file to display. Defaults to 100 with no selector, showing all log lines otherwise 10, if a selector is provided.")
+	cmd.Flags().DurationVar(&flags.sinceSeconds, "since", flags.sinceSeconds, "Only return logs newer than a relative duration like 5s, 2m, or 3h. Defaults to all logs. Only one of since-time / since may be used.")
+	cmd.Flags().StringVar(&flags.sinceTime, "since-time", flags.sinceTime, "Only return logs after a specific date (RFC3339). Defaults to all logs. Only one of since-time / since may be used.")
+	cmd.Flags().BoolVar(&flags.elasticsearch, "elasticsearch", flags.elasticsearch, "Show logs from elasticsearch")
```

Comment:
The help does not state what is happening when the flag is not set.
`Show logs from elasticsearch` -> `If the flag is set the logs are retrieved from the elasticsearch, otherwise from the kubelet`, for example.